File: backendworker/NHSOnline.Backend.PfsApi/OrganDonation/Models/OrganDonationWithdrawRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace NHSOnline.Backend.PfsApi.OrganDonation.Models
{
    public class OrganDonationWithdrawRequest
    {
        public string Identifier { get; set; }

        public string NhsNumber { get; set; }

        public Name Name { get; set; }

        public string Gender { get; set; }

        public DateTime? DateOfBirth { get; set; }

        public string AddressFull { get; set; }

        public Address Address { get; set; }

        public string WithdrawReasonId { get; set; }
    }
}

File: bddtests/src/main/kotlin/mocking/organDonation/models/OrganDonationWithdrawRequest.kt
package mocking.organDonation.models

import mocking.data.organDonation.OrganDonationSerenityHelpers
import models.Patient
import utils.set

data class OrganDonationWithdrawRequest(
        var identifier: String,
        var nhsNumber: String,
        var name: Name,
        var gender: String,
        var dateOfBirth: String,
        var addressFull: String,
        var address: Address,
        var WithdrawReasonId: String
) {
    companion object {
        fun withdrawDecision(patient: Patient): OrganDonationWithdrawRequest{
            val withdrawalReasons: KeyValuePair<String, String> = KeyValuePair("3", "Religious Grounds")
            OrganDonationSerenityHelpers.ORGAN_DONATION_WITHDRAWAL_REASON.set(withdrawalReasons.value)

            return OrganDonationWithdrawRequest(
                    identifier = patient.organDonationRegistrationId,
                    nhsNumber = patient.formattedNHSNumber(),
                    name = Name.fromPatient(patient),
                    gender = patient.sex.toString(),
                    dateOfBirth = patient.age.dateOfBirth,
                    addressFull = patient.contactDetails.address.full(),
                    address = Address.fromPatient(patient),
                    WithdrawReasonId = withdrawalReasons.key
            )
        }
    }
}